tools/warning-to-plist/warning_to_plist/WarningToPlist.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python
+# -------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+
+import argparse
+import logging
+import os
+import shutil
+import sys
+
+try:
+    from .converter.clang_tidy.plist_converter import \
+        ClangTidyPlistConverter
+    from .converter.sanitizers.address.plist_converter import \
+        ASANPlistConverter
+    from .converter.sanitizers.memory.plist_converter import \
+        MSANPlistConverter
+    from .converter.sanitizers.thread.plist_converter import \
+        TSANPlistConverter
+    from .converter.sanitizers.ub.plist_converter import \
+        UBSANPlistConverter
+except ValueError:
+    # Attempted relative import in non-package.
+    from converter.clang_tidy.plist_converter import \
+        ClangTidyPlistConverter
+    from converter.sanitizers.address.plist_converter import \
+        ASANPlistConverter
+    from converter.sanitizers.memory.plist_converter import \
+        MSANPlistConverter
+    from converter.sanitizers.thread.plist_converter import \
+        TSANPlistConverter
+    from converter.sanitizers.ub.plist_converter import \
+        UBSANPlistConverter
+
+
+LOG = logging.getLogger('WarningToPlist')
+
+msg_formatter = logging.Formatter('[%(levelname)s] - %(message)s')
+log_handler = logging.StreamHandler()
+log_handler.setFormatter(msg_formatter)
+LOG.setLevel(logging.INFO)
+LOG.addHandler(log_handler)
+
+
+supported_converters = {
+    ClangTidyPlistConverter.TOOL_NAME: ClangTidyPlistConverter,
+    MSANPlistConverter.TOOL_NAME: MSANPlistConverter,
+    ASANPlistConverter.TOOL_NAME: ASANPlistConverter,
+    TSANPlistConverter.TOOL_NAME: TSANPlistConverter,
+    UBSANPlistConverter.TOOL_NAME: UBSANPlistConverter}
+
+
+def output_to_plist(output, parser_type, output_dir, clean=False):
+    """ Creates .plist files from the given output to the given output dir. """
+    plist_converter = supported_converters[parser_type]()
+    messages = plist_converter.parse_messages(output)
+
+    converters = {}
+    for m in messages:
+        if m.path not in converters:
+            converters[m.path] = supported_converters[parser_type]()
+        converters[m.path].add_messages([m])
+
+    if clean and os.path.isdir(output_dir):
+        LOG.info("Previous analysis results in '%s' have been removed, "
+                 "overwriting with current result", output_dir)
+        shutil.rmtree(output_dir)
+
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    for c in converters:
+        file_name = os.path.basename(c)
+        out_file_name = '{0}_{1}.plist'.format(file_name, parser_type)
```

Comment:
When the actual report is in `lib.c` which is used by `a.c` and `b.c` then the .plist files belonging to these source files will be the same. So the reports of `a.c` will be overwritten by those of `b.c`. In order to avoid this the filename should contain some hash of the stack trace.

Reply:
I tried it out and it worked for me. See the implementation: https://github.com/Ericsson/codechecker/blob/3d525ae4e7432cfa4fbca628c1875e7a3021aa17/tools/warning-to-plist/warning_to_plist/WarningToPlist.py#L69

We will parse the messages of the analyzer output and put each message to a dict where the key will be the file path and the value will be a converter and we add message to it.

Later we will iterate over this dict and create plist files based on the key values:
https://github.com/Ericsson/codechecker/blob/3d525ae4e7432cfa4fbca628c1875e7a3021aa17/tools/warning-to-plist/warning_to_plist/WarningToPlist.py#L79-L87

Reply:
Alright, it works for me too. I had separate files for every sanitizer result. This way processing the next file overwrote the previous .plist file with the same name.